public/app/features/alerting/unified/hooks/useAbilities.ts

```diff
@@ -49,53 +58,137 @@ export enum AlertmanagerAction {
   DeleteMuteTiming = 'delete-mute-timing',
 }
 
-export enum AlertSourceAction {
+// this enum lists all of the available actions we can take on a single alert rule
+export enum AlertRuleAction {
+  Duplicate = 'duplicate-alert-rule',
+  View = 'view-alert-rule',
+  Update = 'update-alert-rule',
+  Delete = 'delete-alert-rule',
+  Explore = 'explore-alert-rule',
+  Silence = 'silence-alert-rule',
+  ModifyExport = 'modify-export-rule',
+}
+
+// this enum lists all of the actions we can perform within alerting in general, not linked to a specific
+// alert source, rule or alertmanager
+export enum AlertingAction {
   // internal (Grafana managed)
   CreateAlertRule = 'create-alert-rule',
   ViewAlertRule = 'view-alert-rule',
   UpdateAlertRule = 'update-alert-rule',
   DeleteAlertRule = 'delete-alert-rule',
+  ExportGrafanaManagedRules = 'export-grafana-managed-rules',
+
   // external (any compatible alerting data source)
   CreateExternalAlertRule = 'create-external-alert-rule',
   ViewExternalAlertRule = 'view-external-alert-rule',
   UpdateExternalAlertRule = 'update-external-alert-rule',
   DeleteExternalAlertRule = 'delete-external-alert-rule',
 }
 
-const AlwaysSupported = true; // this just makes it easier to understand the code
-export type Action = AlertmanagerAction | AlertSourceAction;
+// these just makes it easier to read the code :)
+const AlwaysSupported = true;
+const NotSupported = false;
 
+export type Action = AlertmanagerAction | AlertingAction | AlertRuleAction;
 export type Ability = [actionSupported: boolean, actionAllowed: boolean];
 export type Abilities<T extends Action> = Record<T, Ability>;
 
-export function useAlertSourceAbilities(): Abilities<AlertSourceAction> {
-  // TODO add the "supported" booleans here, we currently only do authorization
-
-  const abilities: Abilities<AlertSourceAction> = {
-    // -- Grafana managed alert rules --
-    [AlertSourceAction.CreateAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleCreate)],
-    [AlertSourceAction.ViewAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleRead)],
-    [AlertSourceAction.UpdateAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleUpdate)],
-    [AlertSourceAction.DeleteAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleDelete)],
-    // -- External alert rules (Mimir / Loki / etc) --
-    // for these we only have "read" and "write" permissions
-    [AlertSourceAction.CreateExternalAlertRule]: [
+/**
+ * This one will check for alerting abilities that don't apply to any particular alert source or alert rule
+ */
+export const useAlertingAbilities = (): Abilities<AlertingAction> => {
+  const canExportGMARules =
+    ctx.hasPermission(AccessControlAction.AlertingProvisioningRead) ||
```

Comment:
Why are we changing the permissions in there?. For exporting users only need `AccessControlAction.AlertingRuleRead`.

Reply:
Ok so it turns out after a conversation that there are slight inconsistencies between provisioning permissions of alert rules and Alertmanager entities. I'm reverting the check back for now to be consistent with the backend permission checks but we should sort those out at some point.